Automattic/newspack-popups
# feat(newsletters): handle subscribe form block event

### All Submissions:






### Changes proposed in this Pull Request:



Handles newsletter subscriptions completed via the [Newspack Newsletters Subscribe block](https://github.com/Automattic/newspack-newsletters/pull/873) for segmentation purposes. Logs a `subscription` event when a contact is successfully added to the connected ESP, meaning the reader will subsequently be segmented as a subscriber.

Also proposes a slightly simpler API for directly adding arbitrary events via the Campaigns API by `POST`ing an array of event objects in a `reader_events` param.

Closes #913.

### How to test the changes in this Pull Request:

1. Publish a prompt segmented to subscribers only.
2. Add a Subscribe form block anywhere on the page (in post content, widget, prompt, etc.).
3. In a new incognito session, subscribe via the form block.
4. Navigate through the site and confirm that you see the prompt for subscribers only.
5. Examine the `wp_newspack_campaigns_reader_events` table and confirm that a new `subscription` event row was logged with pertinent info about the contact. (From WP CLI: `wp db query "SELECT * from wp_newspack_campaigns_reader_events WHERE type = 'subscription' ORDER BY date_created DESC;"`).

### Other information:
